standalone/runtime/src/lib.rs

```diff
@@ -1404,6 +1406,46 @@ impl_runtime_apis! {
             let balance = BalanceWrapper::<Balance> { amount };
             Ok(balance)
         }
+
+        fn estimate_escrow_reward_rate(
+            account_id: AccountId,
+            amount: Option<Balance>,
+            lock_time: Option<BlockNumber>,
+            after: BlockNumber,
+        ) -> Result<UnsignedFixedPoint, DispatchError> {
+            // withdraw previous rewards
+            <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::withdraw_reward(&(), &account_id, NATIVE_CURRENCY_ID)?;
+            // increase amount and/or lock_time
+            Escrow::round_height_and_deposit_for(&account_id, amount.unwrap_or_default(), lock_time.unwrap_or_default())?;
+            // distribute rewards accrued over block count
+            let reward = EscrowAnnuity::min_reward_per_block().saturating_mul(after.into());
+            <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::distribute_reward(&(), NATIVE_CURRENCY_ID, reward)?;
+            let received = <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::compute_reward(&(), &account_id, NATIVE_CURRENCY_ID)?;
```

Comment:
Setting state isn't required for estimating the APR. Something like this would do it:
```rust
let reward = EscrowAnnuity::min_reward_per_block().saturating_mul(after.into());
let total_stake = <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::get_total_stake(&())?;
let user_stake = <EscrowRewards as reward::RewardsApi<(), AccountId, Balance>>::get_stake(&(), NATIVE_CURRENCY_ID)?;
let projected_user_reward = reward * user_stake / total_stake;
let total_locked = Escrow::locked_balance(&account_id).amount;
Ok(UnsignedFixedPoint::checked_from_rational(projected_user_reward, total_locked).unwrap())
```

Reply:
How is `projected_user_reward` different from what I have now?

Reply:
The only difference in that state isn't being set - however you mentioned sandboxing, so if that's possible then your approach is much more elegant